### ISSUE TITLE
Cover block: Nested block font colour is not adjusted properly on style variation switch.

### ISSUE THREAD

**Issue by alaczek** [Automattic/wp-calypso#68465]
### Quick summary

I'm working on the Lynx theme and adding global style variations specifically. I noticed that when switching between variations when going from one with a light background to one with dark background the heading/paragraph color inside a Cover block isn't being automatically adjusted.

### Steps to reproduce

1. Activate the Lynx theme on a test site (easiest way is to go through [Link in Bio flow](https://wordpress.com/setup/patterns?flow=link-in-bio) and create a new site - Lynx will be automatically activated)
2. Once you land in the editor, replace the content with the following pattern:
```
<!-- wp:cover {"overlayColor":"background","minHeight":100,"minHeightUnit":"vh","align":"full"} -->
<div class="wp-block-cover alignfull" style="min-height:100vh"><span aria-hidden="true" class="wp-block-cover__background has-background-background-color has-background-dim-100 has-background-dim"></span><div class="wp-block-cover__inner-container"><!-- wp:group {"layout":{"type":"constrained"},"style":{"spacing":{"blockGap":"var:preset|spacing|20"}}} -->
<div class="wp-block-group"><!-- wp:site-logo {"align":"center","className":"is-style-rounded"} /-->

<!-- wp:site-title {"textAlign":"center","isLink":false,"style":{"typography":{"fontStyle":"normal","fontWeight":"700"}},"fontSize":"x-large"} /-->

<!-- wp:site-tagline {"textAlign":"center","fontSize":"small"} /-->

<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"center","orientation":"horizontal"},"style":{"spacing":{"margin":{"top":"var:preset|spacing|60"}}}} -->
<div class="wp-block-buttons" style="margin-top:var(--wp--preset--spacing--60)"><!-- wp:button {"width":100,"className":"is-style-fill"} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100 is-style-fill"><a class="wp-block-button__link wp-element-button">Must read: The Place That Made Me</a></div>
<!-- /wp:button -->

<!-- wp:button {"width":100,"className":"is-style-fill"} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100 is-style-fill"><a class="wp-block-button__link wp-element-button">Rain Journal by Richard Johnson</a></div>
<!-- /wp:button -->

<!-- wp:button {"width":100,"className":"is-style-fill"} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100 is-style-fill"><a class="wp-block-button__link wp-element-button">10th Day of Rain</a></div>
<!-- /wp:button -->

<!-- wp:button {"width":100,"className":"is-style-fill"} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100 is-style-fill"><a class="wp-block-button__link wp-element-button">Blurring the Lines Between Architecture and Nature</a></div>
<!-- /wp:button -->

<!-- wp:button {"width":100,"className":"is-style-fill"} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100 is-style-fill"><a class="wp-block-button__link wp-element-button">Instant Classic: A story of Persistence</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group --></div></div>
<!-- /wp:cover -->
```
3. Open global styles and switch to "Green" or "Dark Navy".
4. The title/description in the pattern should change color to white in the editor.
<img width="701" alt="image" src="https://user-images.githubusercontent.com/4550351/193196774-344a5005-c6c9-4e8d-b065-c99e741a178d.png">

5. Preview the site. Note the background is changed according to style variation, but the title/description is unreadable.
<img width="793" alt="image" src="https://user-images.githubusercontent.com/4550351/193196809-3c138e71-0a27-44c5-8175-f1b327f16dd0.png">

### What you expected to happen

I expected the preview to be the same as the editor view.

### What actually happened

The title\description is unreadable in the preview (and does not match the editor).

### Browser

Google Chrome/Chromium

### Platform (Simple, Atomic, or both?)

Simple

### Other notes

I wasn't able to replicate it on a local self-hosted installation. 

As best as I can tell the parent cover block has an `is-light` class added (which is what is responsible for automatically adjusting the text color inside), but it's not taking into account the fact that this particular style variation uses dark background color. 
<img width="531" alt="image" src="https://user-images.githubusercontent.com/4550351/193197241-1a6029c9-6e96-4c1f-a346-de41672d19eb.png">

This _might_ be related to #59434

### Reproducibility

Consistent

**Comment by ClassicRKR27** [comment]
- [ ] I have a similar report here: 37490274-hc

Not sure if it's the exact same issue, but this instance was using the Blank Canvas theme.

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Simple – Could Not Replicate

📌 FINDINGS/SCREENSHOTS/VIDEO
- Trying to use the content above is more confusing at this point, so just trying to repro this through the global styles like a new user on an LiB theme might.
- Activated a dark style variation of Barnett first, then opened the editor and changed to different style variations.
- The blocks nested inside the cover block, including a Site Title and buttons, all switched colors with the style variation change, as expected.

#### Screenshots/Recordings

https://user-images.githubusercontent.com/27249804/236014294-dc2a7a35-23a7-42b1-8065-a5dff9a88aa2.mp4

📌 ACTIONS
- Closed as no longer relevant/not reproducible
